backend/games/battle_mogs/src/algorithm.rs

```diff
@@ -438,33 +427,35 @@ impl Generation {
 		input_rarity_1: RarityType,
 		input_generation_2: MogwaiGeneration,
 		input_rarity_2: RarityType,
-		random_hash: &[u8],
+		random_hash: &[u8; 32],
 	) -> (RarityType, MogwaiGeneration, RarityType) {
-		let mut resulting_gen = MogwaiGeneration::default();
-		let mut resulting_rarity = RarityType::default();
-
-		if random_hash.len() >= 12 {
-			let base_rarity = (input_rarity_1 as u16 + input_rarity_2 as u16).saturating_sub(2) / 2;
-
-			let slice = unsafe { &*(&random_hash[0..6] as *const [u8] as *const [u8; 6]) };
-			let (out_rarity_1, out_gen_1) =
-				Self::compute_next_generation_and_rarity(input_generation_1, input_rarity_1, slice);
+		let base_rarity = (input_rarity_1 as u16 + input_rarity_2 as u16).saturating_sub(2) / 2;
+
+		let mut rarity_1_input_hash: [u8; 6] = [0u8; 6];
+		rarity_1_input_hash.copy_from_slice(&random_hash[0..6]);
```

Comment:
This is the only diff except for the indentation. Remove the unsafe block by using a copy_from_slice